tianshou/utils/string.py

```diff
@@ -0,0 +1,536 @@
+"""Copy of sensai.util.string from sensAI commit d7b4afcc89b4d2e922a816cb07dffde27f297354."""
```

Comment:
I think it's better to directly use sensAI utils instead of copy-paste here, otherwise you need to add extra tests and need to pay extra attention to sync the code. Same as logging.py

Reply:
Also the typing is missing for some function

Reply:
With commit fdb0eba93d81fa5e698770b4f7088c87fc1238da we now depend on sensAI instead of copying modules from it.

Reply:
@Trinkle23897, so do you really prefer the added dependencies?

Reply:
For now I admit I hesitate to add this dependency. The drawback I can foresee is that in the future if we want to customize the utility function (e.g., redirect log to some other places) it would be much harder because we need to change sensai as well, and that might slow down the development speed. If you don't mind feel free to revert to the old version, I'm very sorry.

Reply:
I have reverted the commit that made the change. Should we change our minds later, we can always go back (and revert the reversion) - provided that we don't squash the PR.

Reply:
Note that lack of customizability is not an issue though, because the logging module is fully configurable and essentially makes no project-specific assumptions.